fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/index/PerceptionGrid.java

```diff
@@ -47,8 +48,9 @@ public PerceptionGrid(CartesianRectangle bounds, double cellWidth, double cellHe
     }
 
     @Override
-    public List<VehicleObject> getVehiclesInRange(PerceptionRange searchRange) {
-        return vehicleGrid.getItemsInBoundingArea(searchRange.getBoundingBox(), searchRange::isInRange);
+    public List<VehicleObject> getVehiclesInRange(PerceptionModuleOwner owner, PerceptionModel searchRange) {
+        List<VehicleObject> initiallyPerceivedVehicles = vehicleGrid.getItemsInBoundingArea(searchRange.getBoundingBox(), searchRange::isInRange);
+        return searchRange.applyPerceptionModifiers(owner, initiallyPerceivedVehicles);
```

Comment:
the index itself should not apply the perception modifier. its task is to retrieve vehicles within the given range. It would be better if the CentralPerceptionComponent, which calls `getVehiclesInRange`, would apply the perception modifiers after retrieving the plain list of vehicles in range.

Reply:
Good point! 
`getVehiclesInRange` is called in the `SimplePerceptionModule` though, but I think applying the modifiers there is still a lot cleaner